dask/distributed/scheduler.py

```diff
@@ -209,6 +213,15 @@ def block(self):
         self._listen_to_workers_thread.join()
         self._listen_to_clients_thread.join()
 
+    def _client_registration(self, header, payload):
+        """ Client comes in, register it, send back info about the cluster"""
+        payload = pickle.loads(payload)
+        address = header['address']
+        self.clients[address] = payload
+        out_header = {}
+        out_payload = {'clients': self.clients, 'workers': self.workers}
```

Comment:
would suggest removing `self.clients` from this.  Not sure we want to publicize clients to each other.